app/presenters/notification_presenter.rb

```diff
@@ -19,6 +19,25 @@ def list(raw_notifications)
       end
     end
 
+    def list_by_cluster_id(raw_notifications, cluster_id)
+      notifications = []
+      raw_notifications.each do |notification|
+        notification.each do |notification_id, attributes|
+          if attributes.key?("cluster_id") && attributes['cluster_id'] == cluster_id
```

Comment:
Don't think we need to check the `attributes.key?` check here, and we can convert this to a guard clause:
`next unless attributes['cluster_id'] == cluster_id`

Reply:
Here also, shouldnt it be 

`next if attributes['cluster_id'] == cluster_id`

I have limited understanding of "unless", so I might be wrong

Reply:
`next` means to proceed immediately to the next iteration of the loop.
So, we proceed immediately to processing the next notification, unless the cluster_id matches.

Reply:
Gotcha, fixed, please check